### DISCUSSION TITLE
detect if article has property by propertyname

### DISCUSSION THREAD

**Discussion by MatthewJRyan** [openstore-ecommerce/OpenStore#162] [Q&A] [answered]
I wanted to use some custom properties to tell the product display template to show certain icons.

i,e,
PropertyGroup = "Standards"
Properties in group: "BS", "ASTM", "FR", etc
add property to articles....

Then in template 
@if  "what??"
{
  <div>whatever icon/logo want to show<div>
}

I'm doing this in ProductDisplayDetail.cshtml
But i dont know how to check if an article has a certain property (by name).

Can this be done? if so how?

**Comment by DNNMonster** [comment]
Categories and properties are both CATEGORY records.  They are linked by the GROUP record.

I believe you're looking at calling the ProductData.GetCategories func which will allow you to get a List of properties per your needs.

It takes a groupref string param that is useful for getting at specific properties.

https://github.com/openstore-ecommerce/OpenStore/blob/master/Components/Product/ProductData.cs

`var myProps = myProduct.GetCategories("my-groupref", true)`

You can iterate the properties or use LINQ to evaluate things once you've got the data.

You could do this in your Razor template or you might do it in a custom token for little more separation and re-usability.

**Reply by MatthewJRyan** [reply]
this wasnt quite what i needed, but did point me in the right direction of the ProductData class.
After reading that class source i did this instead...

```
@if (product.HasProperty("BSEN")) 
{
    <div class="eds_btn eds_btnInfo">BS EN</div>
}
```

**Comment by DNNMonster** [comment, marked as answer]
Awesome.  Thanks for posting it back.
